src/HaroohiePals.Gui/ImGuiController.cs

```diff
@@ -160,390 +124,441 @@ void main()
-                                p[x] = icon[x, y].PackedValue;
-                        }
-                    }
-                }
-            }
+        _fontTexture.Dispose();
+        _shader.Dispose();
+    }
 
-            foreach (var icon in _icons)
-                loadIcon(registeredIcons[icon], icon.ImageBytes);
+    internal void PressInputChar(char keyChar)
+        => _pressedInputChars.Add(keyChar);
 
-            _fontTexture = new GLTexture(pixels, width, height, PixelFormat.Bgra, PixelType.UnsignedByte,
-                TextureWrapMode.Clamp, TextureWrapMode.Clamp);
-            _fontTexture.SetFilterMode(TextureFilterMode.Linear, TextureFilterMode.Linear);
+    private int RegisterIcon(char character, int size)
+    {
+        var io = ImGui.GetIO();
+        return io.Fonts.AddCustomRectFontGlyph(io.Fonts.Fonts[0], character, size, size, size);
+    }
 
-            io.Fonts.SetTexID((IntPtr)_fontTexture.Handle);
+    private void InitIcons(IReadOnlyCollection<ImGuiIconGlyph> icons)
+    {
+        //Default icons
+        _icons.Add(new ImGuiIconGlyph(Resources.Icons.Folder_16x, FontAwesome6.Folder[0], 16));
+        _icons.Add(new ImGuiIconGlyph(Resources.Icons.Document_16x, FontAwesome6.File[0], 16));
+        _icons.Add(new ImGuiIconGlyph(Resources.Icons.ImportFile_16x, FontAwesome6.FileImport[0], 16));
+        _icons.Add(new ImGuiIconGlyph(Resources.Icons.ExportFile_16x, FontAwesome6.FileExport[0], 16));
+        _icons.Add(new ImGuiIconGlyph(Resources.Icons.VSO_RecycleBin_16x, FontAwesome6.TrashCan[0], 16));
+        _icons.Add(new ImGuiIconGlyph(Resources.Icons.Undo_16x, FontAwesome6.ArrowRotateLeft[0], 16));
+        _icons.Add(new ImGuiIconGlyph(Resources.Icons.Redo_16x, FontAwesome6.ArrowRotateRight[0], 16));
+
+        if (icons != null)
+            _icons.AddRange(icons);
+    }
 
-            io.Fonts.ClearTexData();
-        }
+    private void InitFonts()
+    {
+        bool isDefaultFont = true;
 
-        /// <summary>
-        /// Renders the ImGui draw list data.
-        /// </summary>
-        public void Render()
+        foreach (var font in _fonts)
         {
-            if (_frameBegun)
+            AddFont(font, GetImFontConfig());
+
+            // Merge icon font only for the default font (first font)
+            if (isDefaultFont)
             {
-                _frameBegun = false;
-                ImGui.Render();
-                RenderImDrawData(ImGui.GetDrawData());
+                var iconRanges = new ushort[] { FontAwesome6.IconMin, FontAwesome6.IconMax, 0 };
+                AddFont(_iconFont, GetImFontConfig(true), iconRanges);
             }
+
+            isDefaultFont = false;
         }
+    }
 
-        /// <summary>
-        /// Updates ImGui input and IO configuration state.
-        /// </summary>
-        public void Update(GameWindow wnd, float deltaSeconds)
+    private ImFontConfigPtr GetImFontConfig(bool isIconFont = false)
```

Comment:
```suggestion
    private ImFontConfigPtr CreateImFontConfig(bool isIconFont = false)
```
Get usually implies something already exists and you just get it. Because we're dealing with an unmanaged resource here it's even more important to be clear about ownership.